packages/experiments/src/components/CollapsibleSection/CollapsibleSection.state.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { ICollapsibleSectionProps, ICollapsibleSectionStyles } from './CollapsibleSection.types';
+import { createRef } from 'office-ui-fabric-react';
+import { TStateProps } from '../../utilities/createComponent';
+
+export interface ICollapsibleSectionState {
+  collapsed: boolean;
+}
+
+// TODO: these types seem duplicated from createComponent. try to avoid duplication and be aware of circular references with IStateComponent
+// export type ICollapsibleSectionStateProps = ICollapsibleSectionProps & { styles: { [key in keyof ICollapsibleSectionStyles]: string } };
+export type ICollapsibleSectionStateProps =
+  TStateProps<ICollapsibleSectionProps & { styles: { [key in keyof ICollapsibleSectionStyles]: string } }>;
+
+export class CollapsibleSectionState extends React.Component<
+  ICollapsibleSectionStateProps,
+  ICollapsibleSectionState
+  > {
+  private _titleElement = createRef<HTMLElement>();
+
+  constructor(props: ICollapsibleSectionStateProps) {
+    super(props);
+
+    this.state = {
+      collapsed: !!(props.defaultCollapsed === undefined
+        ? props.collapsed
+        : props.defaultCollapsed)
+    };
+  }
+
+  public render(): JSX.Element {
+    const { collapsed } = this.state;
+
+    // TODO: clean this up. is this the right way to merge and pass props on?
+    // TODO: check React 16 deriveStateFromProps
+    // TODO: analyze functions and determine if any should be moved to view
+    const mergedProps = {
+      ...this.props,
+      collapsed,
+      titleElementRef: this._titleElement,
+      onToggleCollapse: this._onToggleCollapse,
+      onKeyDown: this._onKeyDown,
+      onRootKeyDown: this._onRootKeyDown,
+    };
+
+    return this.props.view(mergedProps);
+  }
+
+  private _onRootKeyDown = (ev: React.KeyboardEvent<Element>) => {
+    switch (ev.which) {
+      case 37:
```

Comment:
Do we have a const enum somewhere that list these out?

Reply:
This definitely needs to be cleaned up and made sure it supports RTL too before this component is ready for promotion